lang/syn/src/idl/defined.rs

```diff
@@ -496,7 +496,9 @@ pub fn gen_idl_type(
                 use crate::parser::context::CrateContext;
                 use quote::ToTokens;
 
-                let source_path = proc_macro2::Span::call_site().source_file().path();
+                // If no path was found, just return an empty path and let the find_path function handle it
+                let source_path = proc_macro2::Span::call_site().local_file().unwrap_or(std::path::PathBuf::new());
+    
```

Comment:
Format error in CI: https://github.com/solana-foundation/anchor/actions/runs/14524829367/job/40754178701?pr=3663#step:7:1

Can also use `unwrap_or_default` here:

```suggestion
                let source_path = proc_macro2::Span::call_site()
                    .local_file()
                    .unwrap_or_default();
```

Reply:
I felt like this way is more explicit to us setting it to the empty path in the case where it's None, but in the 2v1 I'll concur

Reply:
All good. I don't really mind it but `clippy` usually complains about it with [`clippy::unwrap_or_default`](https://rust-lang.github.io/rust-clippy/master/index.html#unwrap_or_default).